### PR TITLE
[java/en] Added abstract classes in java

### PR DESCRIPTION
**Pull request by bk2dcradle**
Added abstract classes in java. Explained briefly how they are different from interfaces. Also listed a few Caveats while using them. All through code of course. (I had created a pull request earlier but I closed it, since it had 3 commits, and now I have squashed my commits into 1).

### CODE REVIEW

**Comment by vendethiel** [comment]
You can `git push --force` instead of the closing the PR, next time :-).

File: java.html.markdown [hunk 1]
```diff
@@ -451,6 +451,74 @@ public class ExampleClass extends ExampleClassParent implements InterfaceOne,
...
+
+	// Method can have a body
+	public void eat()
+	{
+		System.out.println("I am an animal and I am Eating.");  
+		// Note: We can access private variable here.
+		age = 30;
+	}
+
+	// No need to initialise, however in an interface
```
Comment (vendethiel) [inline java.html.markdown]: please fix the "initialize"/"initialise" mix-up in this document :).

File: java.html.markdown [hunk 2]
```diff
@@ -451,6 +451,74 @@ public class ExampleClass extends ExampleClassParent implements InterfaceOne,
...
+	{
+		System.out.println("Bark");
+		// age = 30;	==> ERROR!	age is private to Animal
+	}
+
+	// NOTE: You will get an error if you used the 
+	// @Override annotation here, since java doesn't allow
+	// overriding of static methods.
+	// What is happening here is called METHOD HIDING.
+	// Check out this awesome SO post: (http://stackoverflow.com/questions/16313649/)
```
Comment (vendethiel) [inline java.html.markdown]: why the parens?

**Comment by vendethiel** [comment]
Looks good outside from that! :+1:

**Comment by bk2dcradle** [comment]
Removed the parens and changed "initialise" to "initialize". 
I don't know why I put parens. I have removed them now.

**Comment by vendethiel** [comment]
Thank you! :star2:

**Pull request merged**